nbextensions/usability/toc2/main.js

```diff
@@ -307,7 +333,14 @@ var table_of_contents = function () {
           ul= ul.parent();
           while(!ul.is('ul')){ ul= ul.parent(); }
       }
-
+      // Change link id -- append current num_str so as to get a kind of unique anchor 
```

Comment:
m, ok, so the only thing which goes really bad is manually inserted links in the same notebook (e.g. a manual TOC or to jump around)... For links to certain headlines in nbviewer, I would probably also get these in nbviewer, so the URLs there are the old ones and have the same problems as the old solution but still work for the other cases...

Ok, I rest my case :-)